src/timeseriesUpdate.ts

```diff
@@ -0,0 +1,60 @@
+import {Address, BigInt, log} from '@graphprotocol/graph-ts';
+import { Notional, UpdateAssetRate, UpdateETHRate } from '../generated/Notional/Notional';
+
+import { 
+    AssetExchangeRate,
+    EthExchangeRate
+} from '../generated/schema';
+
+import { 
+    getAssetExchangeRateHistoricalData,
+    getEthExchangeRateHistoricalData
+} from './exchange_rates/utils'
+
+/* 
+    Historical data is stored on an hourly basis to keep it lean, this could be changed in the future
+    or be kept as multiple time periods. New values during the timespan updates the hourly value
+*/
+
+function createHourlyId(currencyId: number, timestamp: i32): string {
+    let uniqueHourIndex = timestamp / 3600; // Integer division will always floor result
+
+    return currencyId
+        .toString()
+        .concat(':')
+        .concat(uniqueHourIndex.toString());
+}
+
```

Comment:
What I would do is put a function in here called `handleBlockUpdates` and call your `updateEthExchangeRate...` and `updateAssetExchangeRate...` methods every time `blockNumber % 276 == 0` or something to that effect. There are about 13 seconds between each block so you can assume that 276 blocks is about an hour.